lib/Driver/ToolChains.cpp

```diff
@@ -1287,6 +1287,26 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
   if (!Linker.empty()) {
     Arguments.push_back(context.Args.MakeArgString("-fuse-ld=" + Linker));
   }
+  
+  // Configure the toolchain
+  // By default, use the system clang++ to link.
+  StringRef Clang = "clang++";
+  if (const Arg *A = context.Args.getLastArg(options::OPT_toolchain)) {
```

Comment:
Yeah, I meant that whether or not `-toolchain` did anything was affected by your choice of target. Whether Darwin should switch to using Clang for linking is a whole different issue, and I'd rather not roll that up into this change. Maybe we can just emit a warning if you use `-toolchain` with a Darwin target for now?